### ISSUE TITLE
Interaction resulting in NotFoundError even when handled

### ISSUE THREAD

**Issue by MazeXP** [Nihlus/Remora.Discord#212]
**Description**
When interaction are handled but a persistent entity is available for that interaction type then
an `NotFoundError` is being returned.

**Steps to Reproduce**
1. Run following example: https://github.com/MazeXP/Remora.Discord/tree/interaction-error/Samples/Interactivity
2. Use slash command: `/interactive dropdown`
3. Select color from dropdown of interaction

**Expected Behavior**
No error to be returned as the interaction is handled by `ColourDropdownEntity`.

**Current Behavior**
```
warn: Remora.Discord.Gateway.Services.ResponderDispatchService[0]
      Error in gateway event responder.
One or more errors occurred.
      [0]:      NotFoundError { Message = No persistent data found for an interactive entity. Did you forget to initialize one when sending the original message? }
```

**Library / Runtime Information**
Latest master with .Net 6.0

**Comment by AraHaan** [comment]
Wait what, entities now broke?

**Comment by MazeXP** [comment]
It's an edge case that an error is reported when there actually is not one.
The interaction is still successful.

**Comment by AraHaan** [comment]
Yeah, in my bot it is always an success result.

**Comment by Nihlus** [comment]
This should be solved by the #213.
